internal/pkg/db/db.go

```diff
@@ -104,31 +109,31 @@ func NewBackend(opts ...Option) (*Backend, error) {
 	return backend, nil
 }
 
-// AddDocument adds the protobom Document to the database and annotates with its source data, hash, and format.
-func (backend *Backend) AddDocument(sbomData []byte) (*sbom.Document, error) {
+// AddDocument adds the protobom document to the database and annotates with given annotations.
+func (backend *Backend) AddDocument(sbomData []byte, annotations ...Option) (*sbom.Document, error) {
+	// defer clearing annotations for next run
+	defer func() {
+		backend.Options.Annotations = nil
+	}()
+
 	sbomReader := reader.New()
 
 	document, err := sbomReader.ParseStream(bytes.NewReader(sbomData))
 	if err != nil {
 		return nil, fmt.Errorf("parsing SBOM data: %w", err)
 	}
 
-	hash := sha256.Sum256(sbomData)
+	// Loop over the `annotations` slice and call each one
```

Comment:
Again, "why is it doing this" rather than "what is it doing"

Reply:
Also the variable should probably be renamed to `opts` or `backendOpts` or something

Reply:
Resolved the name, updated the comment. Let me know what you think